cmd/pullrequest-init/api.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/jenkins-x/go-scm/scm"
+	"go.uber.org/zap"
+)
+
+// Handler handles interactions with the GitHub API.
+type Handler struct {
+	client *scm.Client
+
+	repo  string
+	prNum int
+
+	logger *zap.SugaredLogger
+}
+
+// NewHandler initializes a new handler for interacting with GitHub
+// resources.
+func NewHandler(logger *zap.SugaredLogger, client *scm.Client, repo string, pr int) *Handler {
+	return &Handler{
+		logger: logger,
+		client: client,
+		repo:   repo,
+		prNum:  pr,
+	}
+}
+
+// Download fetches and stores the desired pull request.
+func (h *Handler) Download(ctx context.Context) (*Resource, error) {
+	// Pull Request
+	h.logger.Info("finding pr")
+	pr, _, err := h.client.PullRequests.Find(ctx, h.repo, h.prNum)
+	if err != nil {
+		h.logger.Infof("finding pr: %v", err)
+		return nil, err
+	}
+
+	// Statuses
+	h.logger.Info("finding combined status")
+	status, out, err := h.client.Repositories.FindCombinedStatus(ctx, h.repo, pr.Sha)
+	if err != nil {
+		body, _ := ioutil.ReadAll(out.Body)
```

Comment:
Possible nit: Do you need to call `defer out.Body.Close()` here to make sure a connection / file descriptor isn't leaked?

Reply:
I actually don't know! It'd be weird if you needed to do this everytime, since otherwise you'd always have to do this for each call. Added it just in case.